packages/core/test/scripts/severless-cloud-run/CloudRunSpoke.js

```diff
@@ -0,0 +1,209 @@
+    assert.equal(validResponse.res.statusCode, 200); // error code
+    assert.isTrue(validResponse.res.text.includes("End of serverless execution loop - terminating process")); // Final text in monitor loop.
+    assert.isFalse(validResponse.res.text.includes("[error]")); // There should be no error logs in a valid execution.
+    assert.isFalse(validResponse.res.text.includes("[info]")); // There should be no info logs in a valid execution.
+    assert.isTrue(lastSpyLogIncludes(spy, "Process exited correctly"));
+  });
+  it("Cloud Run Spoke can correctly returns errors over http calls", async function() {
+    // Invalid path should error out when trying to run an executable that does not exist
+    const invalidPathBody = {
+      cloudRunCommand: "npx truffle exec packages/INVALID/index.js --network test",
+      environmentVariables: {
+        CUSTOM_NODE_URL: web3.currentProvider.host,
+        POLLING_DELAY: 0,
+        EMP_ADDRESS: emp.address,
+        TOKEN_PRICE_FEED_CONFIG: defaultUniswapPricefeedConfig // invalid config that should generate an error
+      }
+    };
+
+    const invalidPathResponse = await sendRequest(invalidPathBody);
+    assert.equal(invalidPathResponse.res.statusCode, 500); // error code
+    // Expected error text from an invalid path
+    assert.isTrue(invalidPathResponse.res.text.includes("no such file or directory")); // Check the HTTP response.
+    assert.isTrue(lastSpyLogIncludes(spy, "no such file or directory")); // Check the process logger contained the error.
+    assert.isTrue(lastSpyLogIncludes(spy, "Process exited with error")); // Check the process logger contains exit error.
+
+    // Invalid config should error out before entering the main while loop in the bot.
+    const invalidConfigBody = {
+      cloudRunCommand: "npx truffle exec packages/monitors/index.js --network test",
+      environmentVariables: {
+        CUSTOM_NODE_URL: web3.currentProvider.host,
+        POLLING_DELAY: 0,
+        // missing EMP_ADDRESS. Should error before entering main while loop.
+        TOKEN_PRICE_FEED_CONFIG: defaultUniswapPricefeedConfig // invalid config that should generate an error
+      }
+    };
+
+    const invalidConfigResponse = await sendRequest(invalidConfigBody);
+    assert.equal(invalidConfigResponse.res.statusCode, 500); // error code
+    // Expected error text from an invalid path
+    assert.isTrue(invalidConfigResponse.res.text.includes("Bad environment variables! Specify an `EMP_ADDRESS`"));
+    assert.isTrue(lastSpyLogIncludes(spy, "Bad environment variables! Specify an `EMP_ADDRESS`")); // Check the process logger contained the error.
+    assert.isTrue(lastSpyLogIncludes(spy, "Process exited with error")); // Check the process logger contains exit error.
+
+    // Invalid price feed config should error out before entering main while loop
+    const invalidPriceFeed = {
+      cloudRunCommand: "npx truffle exec packages/monitors/index.js --network test",
```

Comment:
unfortunately, the tests fail if run using node to run the bot because the [Web3WebsocketClient](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/clients/Web3WebsocketClient.js) does not work with `test` network. I'm hoping that some of your work with generalizing truffle/node execution will solve this @mrice32 ?

Reply:
Also, I've had major issues getting these tests to work in Buidler. For now I've moved the tests out of a path CI hits.

Reply:
Hmm weird. Yeah, it makes sense that you might need truffle exec here then. Maybe we can swap this out (in a separate PR) once https://github.com/UMAprotocol/protocol/pull/1928 gets merged.

Reply:
I've had issues with buidler as well. Do you think it's an address lookup issue, where the second process is seeing a different set of addresses than the calling process?